pkg/reconciler/apiserversource/controller.go

```diff
@@ -101,6 +113,16 @@ func NewController(
 		Handler:    controller.HandleAll(impl.EnqueueControllerOf),
 	})
 
+	roleInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.FilterController(&v1.ApiServerSource{}),
```

Comment:
The `FilterController` filters based on the owner references. Therefor we need to set the OwnerReference in the role (and rolebinding) to the source. 
Check the receive_adapter for an example, how to do this easily with the `kmeta` package:
https://github.com/knative/eventing/blob/35c43b6a9156f261ca0bd8480efb01808d1b0676/pkg/reconciler/apiserversource/resources/receive_adapter.go#L68-L70